core/services/wifi/WifiManager.py

```diff
@@ -137,12 +137,12 @@ async def add_network(self, credentials: WifiCredentials) -> int:
         try:
             data = await self.wpa.send_command_add_network()
 
-            network_number = data.decode("utf-8")
+            network_number = int(data)
```

Comment:
Yes. ADD_NETWORK always return an int representing the index of the network on the conf file or a FAIL. Since we already cover the FAIL case, it will always return bytes directly convertible to ints.

Reply:
So send_command_add_network should return an int, not a bytearray